contrib/src/main/java/org/apache/apex/malhar/contrib/kudu/AbstractKuduOutputOperator.java

```diff
@@ -0,0 +1,666 @@
+    checkNotNull(kuduTable,"Kudu Table cannot be null");
+    checkNotNull(kuduSession, "Kudu session cannot be null");
+    allColumnDefs = new HashMap();
+    primaryKeyColumnNames = new HashSet<>();
+    kuduColumnBasedGetters = new HashMap();
+    buildGettersForPojoPayload();
+    isInSafeMode = false;
+    isInReconcilingMode = false;
+    reconcilingWindowId = Stateless.WINDOW_ID;
+    if ( (context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) != Stateless.WINDOW_ID) &&
+        context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) <
+        windowDataManager.getLargestCompletedWindow()) {
+      isInSafeMode = true;
+      reconcilingWindowId = windowDataManager.getLargestCompletedWindow() + 1;
+      isInReconcilingMode = false;
+    }
+  }
+
+  @Override
+  public void deactivate()
+  {
+    try {
+      apexKuduConnection.close();
+    } catch (Exception e) {
+      LOG.error("Could not close kudu session and resources because " + e.getMessage(), e);
+    }
+  }
+
+  @Override
+  public void beforeCheckpoint(long l)
+  {
+    // Nothing to be done here. Child classes can use this if required
+  }
+
+  @Override
+  public void checkpointed(long l)
+  {
+    // Nothing to be done here. Child classes can use this if required
+  }
+
+  @Override
+  public void committed(long windowId)
+  {
+    try {
+      windowDataManager.committed(windowId);
+    } catch (IOException e) {
+      LOG.error("Error while committing the window id " + windowId + " because " + e.getMessage());
+      throw new RuntimeException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    super.setup(context);
+    windowDataManager = getWindowDataManager();
+    if ( windowDataManager == null) {
+      windowDataManager = new FSWindowDataManager();
+    }
+    windowDataManager.setup(context);
+  }
+
+  @Override
+  public void teardown()
+  {
+    super.teardown();
+    windowDataManager.teardown();
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    currentWindowId = windowId;
+    if ( currentWindowId != Stateless.WINDOW_ID) {
+      if (currentWindowId > reconcilingWindowId) {
+        isInSafeMode = false;
+        isInReconcilingMode = false;
+      }
+      if (currentWindowId == reconcilingWindowId) {
+        isInReconcilingMode = true;
+        isInSafeMode = false;
+      }
+      if (currentWindowId < reconcilingWindowId) {
+        isInReconcilingMode = false;
+        isInSafeMode = true;
+      }
+    }
+    numDeletedInThisWindow = 0;
+    numInsertsInThisWindow = 0;
+    numUpdatesInThisWindow = 0;
+    numUpsertsInThisWindow = 0;
+  }
+
+  @Override
+  public void endWindow()
+  {
+    super.endWindow();
+    try {
+      kuduSession.flush();
+    } catch (KuduException e) {
+      LOG.error("Could not flush kudu session on an end window boundary " + e.getMessage(), e);
+      throw new RuntimeException(e.getMessage());
```

Comment:
you can use entire exception instead of just e.getMessage while creating RuntimeException.